.github/workflows/sidecar-build-push.yaml

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+name: Sidecar Build/Push
+on:
+  push:
+    paths:
+      - 'sidecars/**'
+    branches:
+      - master
+
+jobs:
+  build-push:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Clone source code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```

Comment:
> Hello, it is not clear to me what will be the resulting image-registry image when sidecars are added
> 
> I mean, image registry build is a separate github action job. (there is no dependency)
> 
> so it will build plug-in registry image and build modified sidecars
> 
> but then, what will be the sidecar referenced in the plug-in registry image ?
> 
> usecase:
> I provide an updated plug-in yaml with an updated sidecar image
> 
> it will build the sidecar and publish it but probably sidecar image will have the previous link ?

This PR is only about having sidecar definitions in the registry. Plugin registry automation and building will be handled in a separate PR at a later time and will iterate over what is contributed in this PR.

Reply:
There is an undocumented behaviour in the GitHub checkout action, where on a PR it creates an invisible merge commit. If you do not download all branches/refs you cannot compare to the target branch properly.